WireUI/Sources/WireUIBase/MainTabBarController/MainTabBarController.swift

```diff
@@ -0,0 +1,128 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SwiftUI
+import WireDesign
+
+/// A subclass of `UITabBarController` which preconfigures its `viewControllers` property to match
+/// ``MainTabBarController.Tab``'s cases. After initialization each tab contains an empty navigation controller.
+public final class MainTabBarController: UITabBarController {
+
+    public enum Tab: Int, CaseIterable {
+        case contacts, conversations, folders, archive
+    }
+
+    // MARK: - Tab subscript
+
+    public subscript(tab tab: Tab) -> UINavigationController { viewControllers![tab.rawValue] as! UINavigationController }
+
+    // MARK: - Life Cycle
+
+    public required init() {
+        super.init(nibName: nil, bundle: nil)
+        setupTabs()
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) is not supported")
+    }
+
+    private func setupTabs() {
+        viewControllers = Tab.allCases.map { _ in UINavigationController() }
+
+        for tab in Tab.allCases {
+            let tabBarItem: UITabBarItem
+            switch tab {
+            case .contacts:
+                tabBarItem = .init(
+                    title: NSLocalizedString("tabBar.contacts.title", bundle: .module, comment: ""),
+                    image: .init(resource: .TabBar.contactsOutline),
+                    selectedImage: .init(resource: .TabBar.contactsFilled)
+                )
+                tabBarItem.accessibilityIdentifier = "bottomBarPlusButton"
+                tabBarItem.accessibilityLabel = NSLocalizedString("tabBar.contacts.description", bundle: .module, comment: "")
+                tabBarItem.accessibilityHint = NSLocalizedString("tabBar.contacts.hint", bundle: .module, comment: "")
+
+            case .conversations:
+                tabBarItem = .init(
+                    title: NSLocalizedString("tabBar.conversations.title", bundle: .module, comment: ""),
+                    image: .init(systemName: "text.bubble"),
+                    selectedImage: .init(systemName: "text.bubble.fill")
+                )
+                tabBarItem.accessibilityIdentifier = "bottomBarRecentListButton"
+                tabBarItem.accessibilityLabel = NSLocalizedString("tabBar.conversations.description", bundle: .module, comment: "")
+                tabBarItem.accessibilityHint = NSLocalizedString("tabBar.conversations.hint", bundle: .module, comment: "")
+
+            case .folders:
+                tabBarItem = .init(
+                    title: NSLocalizedString("tabBar.folders.title", bundle: .module, comment: ""),
```

Comment:
I don't think it's safe setting the title using type `String`. I think we can generate them using `SwiftGen` and use  the `L10n` struct.

Reply:
Using the initialiser for string catalog now.